flojoy/flojoy_cloud.py

```diff
@@ -0,0 +1,413 @@
+                        cls=NumpyEncoder,
+                    )
+                else:
+                    print(
+                        "For ordered pair type, data must be in"
+                        " dictionary form with keys 'x' and 'y'"
+                    )
+                    raise TypeError
+        match dc_type:
+            case "OrderedTriple":
+                if isinstance(data, dict) and "x" in data:
+                    payload = json.dumps(
+                        {
+                            "data": {
+                                "type": "OrderedTriple",
+                                "x": data["x"],
+                                "y": data["y"],
+                                "z": data["z"],
+                            }
+                        },
+                        cls=NumpyEncoder,
+                    )
+                else:
+                    print(
+                        "For ordered triple type, data must be in"
+                        " dictionary form with keys 'x', 'y', and 'z'"
+                    )
+                    raise TypeError
+            case "DataFrame":
+                data = data.to_json()
+                payload = json.dumps({"data": {"type": "DataFrame", "m": data}})
+            case "Matrix":
+                payload = json.dumps(
+                    {"data": {"type": "Matrix", "m": data}}, cls=NumpyEncoder
+                )
+            case "Grayscale":
+                payload = json.dumps(
+                    {"data": {"type": "Grayscale", "m": data}}, cls=NumpyEncoder
+                )
+            case "Scalar":
+                payload = json.dumps(
+                    {"data": {"type": "Scalar", "c": data}}, cls=NumpyEncoder
+                )
+            case "Vector":
+                payload = json.dumps(
+                    {"data": {"type": "Vector", "v": data}}, cls=NumpyEncoder
+                )
+            case "Image":
+                RGB_img = np.asarray(data)
+                red_channel = RGB_img[:, :, 0]
+                green_channel = RGB_img[:, :, 1]
+                blue_channel = RGB_img[:, :, 2]
+
+                if RGB_img.shape[2] == 4:
+                    alpha_channel = RGB_img[:, :, 3]
+                else:
+                    alpha_channel = None
+                payload = json.dumps(
+                    {
+                        "data": {
+                            "type": "Image",
+                            "r": red_channel,
+                            "g": green_channel,
+                            "b": blue_channel,
+                            "a": alpha_channel,
+                        }
+                    },
+                    cls=NumpyEncoder,
+                )
+
+        return payload
+
+    def fetch_dc(self, dc_id):
+        """
+        A method that retrieves DataContainers from the Flojoy cloud.
+        """
+        url = f"https://cloud.flojoy.ai/api/v1/dcs/{dc_id}"
```

Comment:
Could extract `https://cloud.flojoy.ai/api/v1/"` into a constant (maybe call it `BASE_URL`)